core/src/saros/editor/text/TextPositionUtils.java

```diff
@@ -0,0 +1,160 @@
+
+    if (lineNumber == 0) {
+      return position.getInLineOffset();
+    }
+
+    int previousLineEndOffset = StringUtils.ordinalIndexOf(text, lineSeparator, lineNumber);
+
+    if (previousLineEndOffset == -1) {
+      throw new IllegalStateException(
+          "The given text contains fewer lines than specified by the text position");
+    }
+
+    int lineStartOffset = previousLineEndOffset + lineSeparator.length();
+
+    return lineStartOffset + position.getInLineOffset();
+  }
+
+  /**
+   * Calculates the line and offset delta contained in the text.
+   *
+   * <p>Tries to guess the used line separator by checking for Windows (<code>\r\n</code>) or Unix
+   * line separators (<code>\n</code>) in the text.
+   *
+   * @param text the text for which to calculate the deltas
+   * @return a pair containing the line delta as the first/left element and the offset delta as the
+   *     second/right element
+   * @throws NullPointerException if the given text is <code>null</code>
+   * @see #guessLineSeparator(String)
+   */
+  // TODO remove different line separator handling once internal normalization is set up
+  public static Pair<Integer, Integer> calculateDeltas(String text) {
+    Objects.requireNonNull(text, "The given text must not be null");
+
+    String lineSeparator = guessLineSeparator(text);
+
+    return calculateDeltas(text, lineSeparator);
+  }
+
+  /**
+   * Calculates the line and offset delta contained in the text.
+   *
+   * @param text the text for which to calculate the deltas
+   * @param lineSeparator the line separator contained in the text
+   * @return a pair containing the line delta as the first/left element and the offset delta as the
+   *     second/right element
+   * @throws NullPointerException if the given text or line separator is <code>null</code>
+   */
+  public static Pair<Integer, Integer> calculateDeltas(String text, String lineSeparator) {
+    Objects.requireNonNull(text, "The given text must not be null");
+    Objects.requireNonNull(lineSeparator, "The given line separator must not be null");
+
+    if (text.isEmpty()) {
+      return new ImmutablePair<>(0, 0);
+    }
+
+    int lineDelta = StringUtils.countMatches(text, lineSeparator);
+
+    int offsetDelta;
+
+    if (lineDelta == 0) {
+      offsetDelta = text.length();
+
+    } else {
+      int lastLineStart = StringUtils.lastIndexOf(text, lineSeparator) + lineSeparator.length();
+
+      offsetDelta = text.length() - lastLineStart;
+    }
+
+    return new ImmutablePair<>(lineDelta, offsetDelta);
+  }
+
+  /**
+   * Tries to figure out the line separator used in the text by whether it contains the Windows or
+   * Unix line separator.
+   *
+   * @param text the text to check
+   * @return the used line separator or an empty string if no line separator could be found
+   * @see #WINDOWS_LINE_SEPARATOR
+   * @see #UNIX_LINE_SEPARATOR
+   */
+  public static String guessLineSeparator(String text) {
```

Comment:
From what I have heard, MacOS X hasn't been using `\r` as a line separator for a while now. They moved to being POSIX compliant a while ago and have been using UNIX line endings since then. 
(See corresponding [superuser entry](https://superuser.com/questions/439440/did-mac-os-lion-switch-to-using-line-feeds-lf-n-for-line-breaks-instead-of).)